### PR TITLE
177 number table sync adjustments

### LINKED ISSUE #117: Outbound: add from master list
Coming from these issues: 

- https://github.com/openmsupply/remote-server/issues/999
- https://github.com/openmsupply/openmsupply-client/issues/1015

and as requested by a potential client. 
Can we add the functionality to "add from master list" when editing an outbound shipment?
Mobile has this, though mSupply does not.

### LINKED ISSUE #177: Number table sync adjustment
store_ID is required field now when pushing to Number table.
Sync integration test is failing

### PR DESCRIPTION
**Pull request by andreievg**
closes #177

Apart from store_id addition to number table push, also added comments (was showing this to @Chris-Petty  and couldn't answer some thing without a deeper look).

We can potentially simplify `to_number_name` and `parse_number_name`, but thought to leave it, as it works nicely and it's not complicated

### CODE REVIEW

File: server/README.md [hunk 1]
```diff
@@ -187,6 +187,14 @@ cargo test
 cargo test --features postgres
 ```
 
+- To run sync integration test
+
+Uncomment test attribute, and read description in method comment
+
+https://github.com/openmsupply/open-msupply/blob/e6dc29f1ace2fd22abc5e9a17e368a9d871a6f45/server/server/src/sync/integration_tests/remote_sync_integration_test.rs#L130-L131
+
+See test_remote_syncing in remote_sy
```
Comment (mark-prins) [inline server/README.md:196]: `remote_sy`...? should there be more to this line?

**Review by mark-prins — COMMENTED** [review]
👍  LGTM!

The branch name threw me for a bit! ( issue is 177, branch is 117 ) I was trying to pull the wrong one 🙄

File: server/README.md (hunk 1 above)
Comment (andreievg) [reply server/README.md]: thanks 2b4e713ebf05429515e21c58bc04c52a966746ee

**Review by clemens-msupply — APPROVED** [review]
Looks good

**Comment by andreievg** [comment]
> The branch name threw me for a bit! ( issue is 177, branch is 117 ) I was trying to pull the wrong one

Sorry I messed it up, fixed the PR name and link to issue but was a bit harder to fix branch name so kept it

**Pull request merged**